lib/scala/pkg/src/main/scala/org/enso/pkg/Config.scala

```diff
@@ -112,6 +114,7 @@ case class Config(
   maintainers: List[Contact],
   edition: Option[Editions.RawEdition],
   preferLocalLibraries: Boolean,
+  componentGroups: ComponentGroups,
```

Comment:
As noted earlier - in general the `Config` permits unknown fields. If `component-groups` want to ensure no unknown fields are present within them, I would suggest verifying that we will _never_ add anything that could land there. And if we are not sure, I would suggest somehow handling this in such a way that the launcher/PM can still load a project just to resolve its version, even if some incompatible changes happened to these details.

In general, whenever possible, it would be good for Config to be forwards and backwards compatible so that older launcher/PM versions can still work with new engines.